src/pybind/mgr/dashboard/services/cephfs.py

```diff
@@ -55,6 +56,12 @@ def opendir(self, dirpath):
         try:
             d = self.cfs.opendir(dirpath)
             yield d
+        except cephfs.OSError:
+            dirpath = dirpath.decode().replace("/.snap", "")
+            raise DashboardException(code='invalid_dir_permission',
+                                     msg="Unable to open directory: {} due to invalid permissions.\
+                                     Try adding the +x permission to the directory".format(dirpath),
+                                     component='cephfs')
```

Comment:
If you grep the cephfs Python bindings (`src/pybind/cephfs`) for `OSError` exceptions you'll get:
```py
class OSError(Error):
        super(OSError, self).__init__(errno, strerror)
class PermissionError(OSError):
class ObjectNotFound(OSError):
class NoData(OSError):
class ObjectExists(OSError):
class IOError(OSError):
class NoSpace(OSError):
class InvalidValue(OSError):
class OperationNotSupported(OSError):
class WouldBlock(OSError):
class OutOfRange(OSError):
class ObjectNotEmpty(OSError):
class NotDirectory(OSError):
class DiskQuotaExceeded(OSError):
        return OSError(ret, msg)
```
Therefore, you cannot assume that all `OSError` exceptions are permission or directory related. However, you can capture `PermissionError` instead.

Reply:
This was my concern too. FYI, the error on logging was: `cephfs.OSError: opendir failed: Permission denied [Errno 13]`. As `except cephfs.PermissionError: ` does not seem to catch the error how can I verify that an error is a PermissionError?